akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -398,6 +400,11 @@ trait BasicDirectives {
       ctx.request.entity.toStrict(timeout, maxBytes).flatMap { strictEntity =>
         val newCtx = ctx.mapRequest(_.copy(entity = strictEntity))
         inner(())(newCtx)
+      }.recover {
+        case _: TimeoutException =>
+          throw IllegalRequestException(
```

Comment:
We discussed a bit whether throwing an `IllegalRequestException` would be good enough vs. introducing a specific exception or even a rejection so that user code would have more control over how to handle that situation.

One argument for not making it a rejection is that a request that failed `toStrict` cannot really be handled in an alternative way and shouldn't run further through the routing tree in my opinion.

@raboof wdyt?

Reply:
I agree a rejection isn't helpful in this case. A specific exception is not strictly needed but might be nice, that's up to the contributor AFAIC.